### ISSUE TITLE
Extern function calls not showing correct type dialog

### ISSUE THREAD

**Issue by yrp604** [Vector35/binaryninja-api#1769]
I have a linux kernel module with an imported function call `mount_bdev`. 

![image](https://user-images.githubusercontent.com/8716677/84724568-5e689700-af3d-11ea-8ff4-3d40f1f6a7cd.png)

If I select this function call and hit 'y', I expect the 'Type for function...' popup. Instead, it's treated as a data variable:

![image](https://user-images.githubusercontent.com/8716677/84724670-95d74380-af3d-11ea-8a6f-bb5947b41aab.png)

This is true for all the imported functions in this module.

[dmyfs.ko.txt](https://github.com/Vector35/binaryninja-api/files/4783996/dmyfs.ko.txt)

**Comment by yrp604** [comment]
I have found a variant/related bug to this, where all functions are imported as data elements:

![image](https://user-images.githubusercontent.com/8716677/95065972-99423680-06b6-11eb-865f-08783d2bf8e5.png)

Note that function names are purple here as opposed to yellow above, indicating global data as opposed to code. In both binaries I get the wrong type change dialog.

[sstic.ko.txt](https://github.com/Vector35/binaryninja-api/files/5326635/sstic.ko.txt)

**Comment by yrp604** [comment]
I'm guessing this is caused by this: Unsupported ELF relocation type: R_AARCH64_IRELATIVE

**Comment by nshp** [comment]
Also related: There are no Function objects for these extern functions, so I don't think it's possible to set function properties for them. I'd specifically like to be able to override `clobbered_regs` for `__fentry__` (the hook point for `ftrace`), because it's called from every function and it saves/restores (or doesn't touch) _every_ register. This means every function starts like:
```
int64_t bnx2x_ptp_adjtime()

00000710  int64_t rsi_1
00000710  void* rdi_1
00000710  rsi_1, rdi_1 = __fentry__()
```
with all the function parameters being misattributed to `__fentry__`.

**Comment by plafosse** [comment]
Fixed in >= 2.4.2998
